### PR TITLE
Cloth crafting recipes + Digging sand gives sand

### PR DESCRIPTION
**Pull request by BNGP0**
## About The Pull Request
Adds cloth crafting recipes (White jumpsuit, backpack, improvised gauze(stops bleeding but doesn't heal brute), empty sandbags and bedsheet) since it didn't have any recipes before for some reason. Also, digging any type of dirt or sand now gives actual sand that can be added into empty sandbags by hand, smelted into glass, or turned into sandstone. Sandstone can be crafted back into sand
## Why It's Good For The Game
More uses for cloth, more ways to get sand and sandstone (might be useful later), a new more convenient way to fill sandbags (the old one still works too)
## Changelog
:cl:
add: Added crafts for cloth
add: Added improvised gauze (crafted from cloth, stops bleeding but doesn't heal any damage on itself)
qol: Digging dirt or sand now gives sand as sheets which can be smelted into glass or used in sandbags (Filling them with the shovel the old way works too)
code: Added sand sheets (impure silicates aka glass ore are not gone so don't get confused)
/:cl:

### CODE REVIEW

File: code/game/objects/items/stacks/sheets/mineral.dm [hunk 1]
```diff
@@ -93,6 +99,38 @@ GLOBAL_LIST_INIT(iron_recipes, list ( \
...
+			var/obj/item/stack/sheet/glass/new_item = new(usr.loc)
+			new_item.add_to_stacks(usr)
+			visible_message(span_warning("[src] is molten into glass by [user] with the weldingtool."), null, span_warning(" You hear welding."))
+			var/obj/item/stack/sheet/mineral/sand/R = src
+			var/replace = (user.get_inactive_held_item()==R)
+			R.use(1)
+			if (!R && replace)
+				user.put_in_hands(new_item)
+		return
+	..()
```
Comment (ivanmixo) [inline code/game/objects/items/stacks/sheets/mineral.dm:128]: Uhhh calling parent twice?

File: code/game/objects/items/stacks/sheets/mineral.dm [hunk 2]
```diff
@@ -93,6 +99,38 @@ GLOBAL_LIST_INIT(iron_recipes, list ( \
...
+	desc = "A pile of sand that can be smelted into glass or used in sandbags"
+	singular_name = "sand pile"
+	icon = 'icons/obj/mining.dmi'
+	icon_state = "Glass ore"
+	throw_speed = 4
+	throw_range = 5
+	sheettype = "sand"
+
+/obj/item/stack/sheet/mineral/sand/attackby(obj/item/W as obj, mob/user as mob)
+	..()
```
Comment (ivanmixo) [inline code/game/objects/items/stacks/sheets/mineral.dm:115]:
```suggestion
	. = ..()
```

File: code/game/objects/items/stacks/sheets/mineral.dm [hunk 3]
```diff
@@ -16,9 +17,14 @@ Mineral Sheets
 
 GLOBAL_LIST_INIT(sandstone_recipes, list ( \
 	new/datum/stack_recipe("pile of dirt", /obj/machinery/portable_atmospherics/hydroponics/soil, 3, time = 1 SECONDS, max_per_turf = STACK_RECIPE_ONE_PER_TILE, on_floor = TRUE), \
+	new/datum/stack_recipe("sand", /obj/item/stack/sheet/mineral/sand, 1), \
 	new/datum/stack_recipe("sandstone door", /obj/structure/mineral_door/sandstone, 10, max_per_turf = STACK_RECIPE_ONE_PER_TILE, on_floor = TRUE), \
 	))
 
+GLOBAL_LIST_INIT(sand_recipes, list ( \
+	new/datum/stack_recipe("sandstone", /obj/item/stack/sheet/mineral/sandstone, 1), \
```
Comment (ivanmixo) [inline code/game/objects/items/stacks/sheets/mineral.dm:25]: I'm not a fan of this, potential infinite sandstone door spam is not okay.

File: code/game/objects/items/stacks/sheets/mineral.dm [hunk 4]
```diff
@@ -93,6 +99,38 @@ GLOBAL_LIST_INIT(iron_recipes, list ( \
...
+	throw_range = 5
+	sheettype = "sand"
+
+/obj/item/stack/sheet/mineral/sand/attackby(obj/item/W as obj, mob/user as mob)
+	..()
+	if (iswelder(W))
+		var/obj/item/tool/weldingtool/WT = W
+		if(WT.remove_fuel(0,user))
+			var/obj/item/stack/sheet/glass/new_item = new(usr.loc)
+			new_item.add_to_stacks(usr)
```
Comment (ivanmixo) [inline code/game/objects/items/stacks/sheets/mineral.dm:120]:
```suggestion
			new_item.add_to_stacks(user)
```

File: code/game/objects/items/stacks/sheets/mineral.dm [hunk 5]
```diff
@@ -93,6 +99,38 @@ GLOBAL_LIST_INIT(iron_recipes, list ( \
...
+			visible_message(span_warning("[src] is molten into glass by [user] with the weldingtool."), null, span_warning(" You hear welding."))
```
Comment (ivanmixo) [inline code/game/objects/items/stacks/sheets/mineral.dm:121]:
```suggestion
			visible_message(span_warning("[src] is molten into glass by [user] with the weldingtool."), null, span_warning("You hear welding."))
```

File: code/game/objects/items/stacks/sheets/mineral.dm [hunk 6]
```diff
@@ -93,6 +99,38 @@ GLOBAL_LIST_INIT(iron_recipes, list ( \
...
+			var/obj/item/stack/sheet/mineral/sand/R = src
+			var/replace = (user.get_inactive_held_item()==R)
+			R.use(1)
+			if (!R && replace)
```
Comment (ivanmixo) [inline code/game/objects/items/stacks/sheets/mineral.dm:125]: Why not just use `src` instead?

File: code/game/objects/items/stacks/sheets/mineral.dm [hunk 7]
```diff
@@ -93,6 +99,38 @@ GLOBAL_LIST_INIT(iron_recipes, list ( \
...
+	name = "sand pile"
+	desc = "A pile of sand that can be smelted into glass or used in sandbags"
+	singular_name = "sand pile"
+	icon = 'icons/obj/mining.dmi'
+	icon_state = "Glass ore"
+	throw_speed = 4
+	throw_range = 5
+	sheettype = "sand"
+
+/obj/item/stack/sheet/mineral/sand/attackby(obj/item/W as obj, mob/user as mob)
```
Comment (ivanmixo) [inline code/game/objects/items/stacks/sheets/mineral.dm:114]:
```suggestion
/obj/item/stack/sheet/mineral/sand/attackby(obj/item/W, mob/user)
```
ew

File: code/game/objects/items/stacks/sheets/mineral.dm [hunk 8]
```diff
@@ -93,6 +99,38 @@ GLOBAL_LIST_INIT(iron_recipes, list ( \
 	. = ..()
 	recipes = GLOB.sandstone_recipes
 
+// i know there's "sand" in mining module (aka impure silicates aka glass ore), but it doesn't work as a sheet and has some other problems
+// you can remove this if you turn all ores into sheets so they stack, or maybe i'll rework mining module myself
```
Comment (ivanmixo) [inline code/game/objects/items/stacks/sheets/mineral.dm:103]: What? Just replace all instances of this old sand type with your new one and delete this comment.

File: code/game/objects/items/stacks/sheets/mineral.dm [hunk 9]
```diff
@@ -93,6 +99,38 @@ GLOBAL_LIST_INIT(iron_recipes, list ( \
...
+/obj/item/stack/sheet/mineral/sand
+	name = "sand pile"
+	desc = "A pile of sand that can be smelted into glass or used in sandbags"
+	singular_name = "sand pile"
+	icon = 'icons/obj/mining.dmi'
+	icon_state = "Glass ore"
```
Comment (ivanmixo) [inline code/game/objects/items/stacks/sheets/mineral.dm:109]: Are you sure this is a valid icon_state?

**Review by ivanmixo — CHANGES_REQUESTED** [review]

File: code/game/objects/items/stacks/sheets/mineral.dm (hunk 9 above)
Comment (BNGP0) [reply code/game/objects/items/stacks/sheets/mineral.dm:109]: I tested it and it works in game. Although the sprite doesn't change depending on the amount of sheets in the stack.

File: code/game/objects/items/stacks/sheets/mineral.dm (hunk 3 above)
Comment (BNGP0) [reply code/game/objects/items/stacks/sheets/mineral.dm:25]: K i should probably make 1 sandstone worth 10 or 20 sand for ballance

**Comment by Lumipharon** [comment]
we have cloth?

**Comment by FosterGrime-Heymynameslouie** [comment]
now that I think of it, where do you even get cloth?

**Comment by BNGP0** [comment]
> now that I think of it, where do you even get cloth?

I wanted to add the ability to make cloth by cutting jumpsuits like on shiptest later

**Comment by ivanmixo** [comment]
Considering we have an unlimited amount of clothing that does not seem like a good idea

**Comment by DorfGuy1** [comment]
what would even be the issue with having unlimited cloth? Unlimited bedsheets? Unlimited shitty bandages?

**Comment by FosterGrime-Heymynameslouie** [comment]
> what would even be the issue with having unlimited cloth? Unlimited bedsheets? Unlimited shitty bandages?

unlimited sandbags


I think a better alternative to the cloth acquisition is that instead of from jumpsuits we just add the cotton plant
give hydroponics some love and ST's something to do other than stare at dry paint

**Pull request closed without merging**